test/test_compile.py

```diff
@@ -0,0 +1,120 @@
+# This file is part of arduino-cli.
+
+# Copyright 2019 ARDUINO SA (http://www.arduino.cc/)
+
+# This software is released under the GNU General Public License version 3,
+# which covers the main part of arduino-cli.
+# The terms of this license can be found at:
+# https://www.gnu.org/licenses/gpl-3.0.en.html
+
+# You can be released from the requirements of the above licenses by purchasing
+# a commercial license. Buying such a license is mandatory if you want to modify or
+# otherwise use the software for commercial activities involving the Arduino
+# software without disclosing the source code of your own applications. To purchase
+# a commercial license, send an email to license@arduino.cc.
+import pytest
+import json
+import os
+
+from .common import running_on_ci
+
+
+def test_compile_without_fqbn(run_command):
+    # Init the environment explicitly
+    result = run_command("core update-index")
```

Comment:
Here you are mixing testing the "core" command and the compile command. It does not sound good to me. Eventually we should have a setup_test function running a `core update-index` just once. As of now we are running 
```    result = run_command("core update-index")
    assert result.ok

     # # Download latest AVR
    result = run_command("core install arduino:avr")
    assert result.ok```

over and over again. Please remove duplication of code making the tests running significantly slower too.

Reply:
Yep, I agree.. and as said in person last week I'm currently moving the shared testing code into a dedicated fixture in pytest. This is happening in a separated branch with a general review of the integration test suite.